net/dns/manager_linux.go

```diff
@@ -40,6 +44,11 @@ func NewOSConfigurator(logf logger.Logf, interfaceName string) (ret OSConfigurat
 	if err != nil {
 		return nil, err
 	}
+	publishOnce.Do(func() {
+		m := clientmetric.NewGauge(fmt.Sprintf("linux_dns_mode_%s", mode))
+		m.Set(1)
+	})
+	logf("dns: using %q mode", mode)
```

Comment:
There is a log of how it decided what mode to operate in:
`dns: [resolved-ping=yes rc=resolved nm=no ret=systemd-resolved]`
However it seems fine to log it here in a more clear way.